cmake/patches/onnx/onnx.patch

```diff
@@ -469,3 +469,561 @@ index 2c63c910..b0e4c32d 100644
++            [make_node("Einsum", ["x", "y"], ["z"], equation="ij,kj->ik")],
++            [],
++        )
++        self._assert_inferred(graph, [make_tensor_value_info("z", TensorProto.FLOAT, (3, 3))])  # type: ignore
++
++    def test_einsum_row_multiplication_A2_B2(
++        self,
++    ) -> None:  # each row of A2 multiplied by B2
++        graph = self._make_graph(
++            [("x", TensorProto.FLOAT, (3, 3)), ("y", TensorProto.FLOAT, (3, 3))],
++            [make_node("Einsum", ["x", "y"], ["z"], equation="ij,kj->ikj")],
++            [],
++        )
++        self._assert_inferred(graph, [make_tensor_value_info("z", TensorProto.FLOAT, (3, 3, 3))])  # type: ignore
++
++    def test_einsum_value_multiplication_A2_B2(
++        self,
++    ) -> None:  # each value of A2 multiplied by B2
++        graph = self._make_graph(
++            [("x", TensorProto.FLOAT, (3, 3)), ("y", TensorProto.FLOAT, (3, 3))],
++            [make_node("Einsum", ["x", "y"], ["z"], equation="ij,kl->ijkl")],
++            [],
++        )
++        self._assert_inferred(graph, [make_tensor_value_info("z", TensorProto.FLOAT, (3, 3, 3, 3))])  # type: ignore
++
++    def test_einsum_scalar_times_array(self) -> None:  # Scalar times array
++        graph = self._make_graph(
++            [("x", TensorProto.FLOAT, ()), ("y", TensorProto.FLOAT, (3, 3))],
++            [make_node("Einsum", ["x", "y"], ["z"], equation=",ij->ij")],
++            [],
++        )
++        self._assert_inferred(graph, [make_tensor_value_info("z", TensorProto.FLOAT, (3, 3))])  # type: ignore
++
++    def test_einsum_matrix_vector_A2_B1(self) -> None:  # Matrix and vector.
++        graph = self._make_graph(
++            [("x", TensorProto.FLOAT, (3, 3)), ("y", TensorProto.FLOAT, (3,))],
++            [make_node("Einsum", ["x", "y"], ["z"], equation="ij,j->i")],
++            [],
++        )
++        self._assert_inferred(graph, [make_tensor_value_info("z", TensorProto.FLOAT, (3,))])  # type: ignore
++
++    def test_einsum_diag_multiplication_A2_B2(
++        self,
++    ) -> None:  # diagonals multiplied by each other
++        graph = self._make_graph(
++            [("x", TensorProto.FLOAT, (3, 3)), ("y", TensorProto.FLOAT, (3, 3))],
++            [make_node("Einsum", ["x", "y"], ["z"], equation="ii,ii->i")],
++            [],
++        )
++        self._assert_inferred(graph, [make_tensor_value_info("z", TensorProto.FLOAT, (3,))])  # type: ignore
++
++    def test_einsum_diag_dot_product_A2_B2(self) -> None:  # dot product of diagonals
++        graph = self._make_graph(
++            [("x", TensorProto.FLOAT, (3, 3)), ("y", TensorProto.FLOAT, (3, 3))],
++            [make_node("Einsum", ["x", "y"], ["z"], equation="ii,ii->")],
++            [],
++        )
++        self._assert_inferred(graph, [make_tensor_value_info("z", TensorProto.FLOAT, ())])  # type: ignore
++
+     def test_negative_log_likehood_shape_is_NCdd(self) -> None:
```

Comment:
This seems fishy. I don't see it in the original change: https://github.com/onnx/onnx/pull/6010/files

Reply:
These lines won't be considered as git change or diff because there is only `+` icon in front of these lines. This is just extra lines in this patch file and won't be considered for patch.

Reply:
Oh confusing, but I see now the double `++` is the *actual* additions for the patch, and the single `+` is for anchoring I guess.